src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs

```diff
@@ -56,56 +30,35 @@ public static IHealthChecksBuilder AddMongoDb(
     /// <returns>The specified <paramref name="builder"/>.</returns>
     public static IHealthChecksBuilder AddMongoDb(
         this IHealthChecksBuilder builder,
-        string mongodbConnectionString,
-        string mongoDatabaseName,
+        Func<IServiceProvider, IMongoClient>? clientFactory = default,
+        Func<IServiceProvider, string>? databaseNameFactory = default,
         string? name = default,
         HealthStatus? failureStatus = default,
         IEnumerable<string>? tags = default,
         TimeSpan? timeout = default)
     {
         return builder.Add(new HealthCheckRegistration(
             name ?? NAME,
-            sp => new MongoDbHealthCheck(mongodbConnectionString, mongoDatabaseName),
+            sp => Factory(sp, clientFactory, databaseNameFactory),

```

Comment:
@adamsitnik - this will execute the `clientFactory` every time health checks are run. Is that intentional?

Reply:
@eerhardt yes, because some clients may register something else than a singleton. For example a client instance that is valid only for some period of time like in https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/2087